api/v1alpha3/awsmachine_webhook.go

```diff
@@ -184,9 +210,13 @@ func (r *AWSMachine) ValidateDelete() error {
 // Default implements webhook.Defaulter such that an empty CloudInit will be defined with a default
 // SecureSecretsBackend as SecretBackendSecretsManager iff InsecureSkipSecretsManager is unset
 func (r *AWSMachine) Default() {
-	if !r.Spec.CloudInit.InsecureSkipSecretsManager && r.Spec.CloudInit.SecureSecretsBackend == "" {
+	if !r.Spec.CloudInit.InsecureSkipSecretsManager && r.Spec.CloudInit.SecureSecretsBackend == "" && !r.ignitionEnabled() {
 		r.Spec.CloudInit.SecureSecretsBackend = SecretBackendSecretsManager
 	}
+
+	if r.ignitionEnabled() && r.Spec.Ignition.Version == "" {
+		r.Spec.Ignition.Version = "2.3"
```

Comment:
Should this default be documented somewhere?

Reply:
I made it a constant, though controller code still use different value, to be able to eventually update patch versions, which shouldn't have breaking changes. I'm not sure if this should be unified.

There is still duplication in kube-builder annotation: `// +kubebuilder:validation:Enum="2.3"`. Do you know if this can be eliminated somehow?

Reply:
I'll mark it as resolved, as I guess it's fine for now or it should get a fresh comment if this is still a problem.